engine/env/spark/src/main/java/org/datacleaner/spark/utils/HadoopUtils.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.spark.utils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.metamodel.MetaModelException;
+import org.apache.metamodel.util.FileHelper;
+
+import com.google.common.base.Strings;
+
+public class HadoopUtils {
```

Comment:
At the moment, the only thing we/I managed to get working on MapR is a completely manual ```spark-submit``` invocation from within the console on one of the nodes itself. I hope we find a way to also do something as a more external client, but if we don't then at least we should keep it in mind that for MapR users we have some limitations and certain convenience utilities that aren't working.